src/RouterSelector.web.js

```diff
@@ -42,34 +46,23 @@ const handleLinks = async () => {
   try {
     const { magiclink } = params
 
-    if (magiclink) {
-      let userNameAndPWD = Buffer.from(decodeURIComponent(magiclink), 'base64').toString()
-      let userNameAndPWDArray = userNameAndPWD.split('+')
-      log.debug('recoverByMagicLink', { magiclink, userNameAndPWDArray })
-      if (userNameAndPWDArray.length === 2) {
-        const userName = userNameAndPWDArray[0]
-        const userPwd = userNameAndPWDArray[1]
-        const UserStorage = await retryImport(() => import('./lib/gundb/UserStorageClass')).then(_ => _.UserStorage)
-
-        const mnemonic = await UserStorage.getMnemonic(userName, userPwd)
-
-        if (mnemonic && bip39.validateMnemonic(mnemonic)) {
-          const mnemonicsHelpers = retryImport(() => import('./lib/wallet/SoftwareWalletProvider'))
-          const { saveMnemonics } = await mnemonicsHelpers
-          await saveMnemonics(mnemonic)
-          await AsyncStorage.setItem('GD_isLoggedIn', true)
-          fireEvent(SIGNIN_SUCCESS)
-          restart('/')
-        }
-      }
-    } else {
-      let path = window.location.pathname.slice(1)
-      path = path.length === 0 ? 'AppNavigation/Dashboard/Home' : path
-
-      if (params && Object.keys(params).length > 0) {
-        const dest = { path, params }
-        log.debug('Saving destination url', dest)
-        await AsyncStorage.setItem(DESTINATION_PATH, dest)
+    let userNameAndPWD = Buffer.from(decodeURIComponent(magiclink), 'base64').toString()
```

Comment:
duplicated code, need to move out to the separate hook

Reply:
not relevant for now